palette/src/gradient/mod.rs

```diff
@@ -97,6 +129,16 @@ impl<C: Mix + Clone> Gradient<C> {
         min_color.mix(max_color, factor)
     }
 
+    /// Create a gradient of colors with custom spacing and domain. There must
+    /// be at least one color and they are expected to be ordered by their
+    /// position value.
+    pub fn with_domain(colors: T) -> Gradient<C, T> {
+        assert!(!colors.is_empty());
+
+        //Maybe sort the colors?
+        Gradient(colors)
```

Comment:
It could be nice, but I'm not sure what's the better option. On one hand it removes panics from the code, on the other hand it pushes them over to the user's code instead. It's not a complicated validation so I'm still leaning towards not returning an `Option` and sparing people the need to always check it themselves.